### PR TITLE
Other messages new sending

### PR DESCRIPTION
**Pull request by Bilb**
Move the sync message of closed groups open groups and contacts to the new pipeline

### CODE REVIEW

File: js/models/messages.js [hunk 1]
```diff
@@ -1447,39 +1448,40 @@
...
+          destination: this.get('destination'),
+          expirationStartTimestamp: this.get('expirationStartTimestamp'),
+          sent_to: this.get('sent_to'),
+          unidentifiedDeliveries: this.get('unidentifiedDeliveries'),
+        };
+        const sentSyncMessage = new libsession.Messages.Outgoing.SentSyncMessage(
+          sentSyncMessageParams
+        );
+
+        return libsession
```
Comment (Mikunj) [inline js/models/messages.js:1473]: use async/await here

File: libloki/api.js [hunk 2]
```diff
@@ -115,16 +115,13 @@
     // Serialise array of byteBuffers into 1 byteBuffer
     const byteBuffer = serialiseByteBuffers(contactDetails);
     const data = new Uint8Array(byteBuffer.toArrayBuffer());
-    const contacts = new textsecure.protobuf.SyncMessage.Contacts({
+    return new libsession.Messages.Outgoing.ContactSyncMessage({
```
Comment (Mikunj) [inline libloki/api.js:118]: should this message take in `rawContacts` instead and we can build the protobuf inside it?

File: libloki/api.js [hunk 3]
```diff
@@ -141,37 +138,12 @@
     // Serialise array of byteBuffers into 1 byteBuffer
     const byteBuffer = serialiseByteBuffers([groupDetail]);
     const data = new Uint8Array(byteBuffer.toArrayBuffer());
-    const groups = new textsecure.protobuf.SyncMessage.Groups({
+    return new libsession.Messages.Outgoing.ClosedGroupSyncMessage({
```
Comment (Mikunj) [inline libloki/api.js:141]: Same here, should it take in `rawGroups` and do the protobuf inside it?

File: libloki/api.js [hunk 4]
```diff
@@ -196,9 +168,8 @@
   window.libloki.api = {
     sendSessionEstablishedMessage,
     sendSessionRequestsToMembers,
-    createContactSyncProtoMessage,
-    createGroupSyncProtoMessage,
-    createOpenGroupsSyncProtoMessage,
+    createContactSyncMessage,
```
Comment (Mikunj) [inline libloki/api.js:171]: Best to not put stuff in libloki as we're going to remove it in the future anyway. Maybe a file in obsession would be better

File: libtextsecure/sendmessage.js [hunk 5]
```diff
@@ -818,8 +669,15 @@ MessageSender.prototype = {
     const result = await sendPromise;
 
     // Sync the group message to our other devices
-    const encoded = textsecure.protobuf.DataMessage.encode(proto);
-    this.sendSyncMessage(encoded, timestamp, null, null, [], [], options);
+    const sentSyncMessageParams = {
+      timestamp,
+      dataMessage: proto,
+    };
+    const sentSyncMessage = new libsession.Messages.Outgoing.SentSyncMessage(
```
Comment (Mikunj) [inline libtextsecure/sendmessage.js:676]: Wouldn't it be easier if you just made the function take an object instead?
```ts
const sentSyncMessage = new libsession.Messages.Outgoing.SentSyncMessage({
    timestamp,
    dataMessage: proto
});
```

File: ts/session/index.ts [hunk 6]
```diff
@@ -1,6 +1,8 @@
 import * as Messages from './messages';
 import * as Protocols from './protocols';
 import * as Types from './types';
+import * as Utils from './utils';
+
 import { getMessageQueue } from './instance';
```
Comment (Mikunj) [inline ts/session/index.ts:6]:
```suggestion
export * from './instance';
```

File: ts/session/messages/outgoing/content/sync/SentSyncMessage.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import { SyncMessage } from './SyncMessage';
+import { SignalService } from '../../../../../protobuf';
+import { MessageParams } from '../../Message';
+import { PubKey } from '../../../../types';
+
+interface SentSyncMessageParams extends MessageParams {
+  dataMessage: SignalService.DataMessage;
+  expirationStartTimestamp?: number;
+  sentTo?: [PubKey];
```
Comment (Mikunj) [inline ts/session/messages/outgoing/content/sync/SentSyncMessage.ts:9]:
```suggestion
  sentTo?: Array<PubKey>;
```

need to make change in the variable inside `SentSyncMessage` too

File: ts/session/messages/outgoing/content/sync/SentSyncMessage.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  unidentifiedDeliveries?: [PubKey];
```
Comment (Mikunj) [inline ts/session/messages/outgoing/content/sync/SentSyncMessage.ts:10]:
```suggestion
  unidentifiedDeliveries?: Array<PubKey>;
```

need to make change in the variable inside SentSyncMessage too

File: ts/session/messages/outgoing/content/sync/SentSyncMessage.ts (hunk 7 above)
Comment (Bilb) [reply ts/session/messages/outgoing/content/sync/SentSyncMessage.ts:9]: fixed

File: ts/session/messages/outgoing/content/sync/SentSyncMessage.ts (hunk 8 above)
Comment (Bilb) [reply ts/session/messages/outgoing/content/sync/SentSyncMessage.ts:10]: fixed

File: js/models/messages.js (hunk 1 above)
Comment (Bilb) [reply js/models/messages.js:1473]: fixed

File: ts/session/index.ts (hunk 6 above)
Comment (Bilb) [reply ts/session/index.ts:6]: fixed

File: libloki/api.js (hunk 2 above)
Comment (Bilb) [reply libloki/api.js:118]: best would be for it to take the `sessionContacts` directly, I will check what I can do

File: libtextsecure/sendmessage.js (hunk 5 above)
Comment (Bilb) [reply libtextsecure/sendmessage.js:676]: I just separated the creation of the object from the function call itself, for improved readability

File: libloki/api.js (hunk 2 above)
Comment (Mikunj) [reply libloki/api.js:118]: Well not necessarily, a `RawContact` can just be the information needed for contact sync, the message shouldn't need to worry about how to extract that information from `sessionContacts`

File: ts/session/messages/outgoing/content/sync/ContactSyncMessage.ts [hunk 9]
```diff
@@ -1,23 +1,63 @@
...
-    this.data = params.data;
+    this.rawContacts = params.rawContacts;
   }
 
   protected syncProto(): SignalService.SyncMessage {
     const syncMessage = super.syncProto();
+    const contactsWithVerified = this.rawContacts.map(c => {
+      let protoState = SignalService.Verified.State.DEFAULT;
+
+      if (c.verifiedStatus === 1 || c.verifiedStatus === 2) {
```
Comment (Mikunj) [inline ts/session/messages/outgoing/content/sync/ContactSyncMessage.ts:31]:
```suggestion
      if (c.verifiedStatus > 0) {
```

File: libloki/api.js (hunk 4 above)
Comment (Bilb) [reply libloki/api.js:128]: we will move this out after the refactor is done

**Pull request merged**